src/components/PopupWithConfirmation.js

```diff
@@ -0,0 +1,29 @@
+import Popup from "./Popup.js";
+
+export default class PopupWithConfirmation extends Popup {
+  constructor({ popupSelector }) {
+    super({ popupSelector });
+    this._modalForm = this._popupElement.querySelector(".modal__form");
+    this._submitButton = this._popupElement.querySelector(".modal__button");
+  }
+
+  setSubmitText(submit, submitText = "Saving...") {
```

Comment:
Great that you've implemented the code for showing loading via the class methods. It's the best way!